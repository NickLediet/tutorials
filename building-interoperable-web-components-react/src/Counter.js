
class Counter extends HTMLElement {
    connectedCallback() {
        this.innerHTML = `
            <div style="color: green">Hey there</div>
        `
    }
}

if (!customElements.get("counter-wc")) {
    customElements.define("counter-wc", Counter);
}