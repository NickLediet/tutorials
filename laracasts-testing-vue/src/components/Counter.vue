<template>
    <div class="counter">
        <h1 class="counter__text">
            Current count is: 
            <span class="counter__count">{{count}}</span>
        </h1>
        <button 
            @click="increment"
            class="counter__add">+</button>
        <button 
            @click="decrement"
            class="counter__decrement">-</button>

    </div>
</template>
<script>
export default {
    name: 'counter',
    data() {
        return {
            count: 0
        }
    },
    methods: {
        increment() {
            this.count++;
        },
        decrement() {
            this.count = (this.count > 0) ? 
                (this.count - 1) : 0;
        }
    }
}
</script>