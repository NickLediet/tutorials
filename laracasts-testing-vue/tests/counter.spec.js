import {describe, it} from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Counter from '../src/components/Counter.vue';



describe('Counter.vue', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(Counter);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('defaults to a count of zero', () => {
        const count = wrapper.vm.count;

        expect(count).toBe(0);
    })

    it('presents the current count', () => {
        const count = wrapper.vm.count;
        const countEl = wrapper.find('.counter__count')

        expect(parseInt(countEl.text())).toBe(count);
    });

    it('should increment the value when increased', () => {
       const addButtonEl = wrapper.find('.counter__add');

       addButtonEl.trigger('click');
       addButtonEl.trigger('click');
      
       // The count value should have incremented twice
       expect(wrapper.vm.count).toBe(2);
    });

    it('should decrease the value when decremented', () =>  {
        const decrementButtonEl = wrapper.find('.counter__decrement');
        const addButtonEl = wrapper.find('.counter__add');
        // Bring the count up to 2
        addButtonEl.trigger('click');
        addButtonEl.trigger('click');
        // Decrement it to 1
        decrementButtonEl.trigger('click');

        expect(wrapper.vm.count).toBe(1);
    });

    it('should not decrease past zero', () => {
        const decrementButtonEl = wrapper.find('.counter__decrement');

        decrementButtonEl.trigger('click');
        decrementButtonEl.trigger('click');
        decrementButtonEl.trigger('click');

        expect(wrapper.vm.count).toBe(0);
    })
});