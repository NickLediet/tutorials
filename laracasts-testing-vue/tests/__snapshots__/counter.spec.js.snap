// Vitest Snapshot v1

exports[`Counter.vue > should render 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="counter"
      >
        <h1
          class="counter__text"
        >
           Current count is: 
          <span
            class="counter__count"
          >
            0
          </span>
        </h1>
        <button
          class="counter__add"
        >
          +
        </button>
        <button
          class="counter__decrement"
        >
          -
        </button>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "count": 0,
          "decrement": [Function],
          "hasOwnProperty": [Function],
          "increment": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "count": 2,
            "decrement": 4,
            "hasOwnProperty": 4,
            "increment": 4,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "count": 0,
            "decrement": [Function],
            "hasOwnProperty": [Function],
            "increment": [Function],
          },
          "da": null,
          "data": {
            "count": 0,
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {},
          "propsDefaults": {},
          "propsOptions": [
            {},
            [],
          ],
          "provides": {},
          "proxy": {
            "count": 0,
            "decrement": [Function],
            "hasOwnProperty": [Function],
            "increment": [Function],
          },
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": {},
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": " Current count is: ",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el":  Current count is: ,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "0",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <span
                      class="counter__count"
                    >
                      0
                    </span>,
                    "key": null,
                    "patchFlag": 1,
                    "props": {
                      "class": "counter__count",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "span",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <h1
                  class="counter__text"
                >
                   Current count is: 
                  <span
                    class="counter__count"
                  >
                    0
                  </span>
                </h1>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "counter__text",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "h1",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "+",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <button
                  class="counter__add"
                >
                  +
                </button>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "counter__add",
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "-",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <button
                  class="counter__decrement"
                >
                  -
                </button>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "counter__decrement",
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "0",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <span
                  class="counter__count"
                >
                  0
                </span>,
                "key": null,
                "patchFlag": 1,
                "props": {
                  "class": "counter__count",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "span",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="counter"
            >
              <h1
                class="counter__text"
              >
                 Current count is: 
                <span
                  class="counter__count"
                >
                  0
                </span>
              </h1>
              <button
                class="counter__add"
              >
                +
              </button>
              <button
                class="counter__decrement"
              >
                -
              </button>
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": {
              "class": "counter",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "C:/Users/nicholas.lediet/Projects/tutorials/laracasts-testing-vue/src/components/Counter.vue",
            "components": {},
            "data": [Function],
            "methods": {
              "decrement": [Function],
              "increment": [Function],
            },
            "name": "counter",
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="counter"
        >
          <h1
            class="counter__text"
          >
             Current count is: 
            <span
              class="counter__count"
            >
              0
            </span>
          </h1>
          <button
            class="counter__add"
          >
            +
          </button>
          <button
            class="counter__decrement"
          >
            -
          </button>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "C:/Users/nicholas.lediet/Projects/tutorials/laracasts-testing-vue/src/components/Counter.vue",
          "components": {},
          "data": [Function],
          "methods": {
            "decrement": [Function],
            "increment": [Function],
          },
          "name": "counter",
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="counter"
        >
          <h1
            class="counter__text"
          >
             Current count is: 
            <span
              class="counter__count"
            >
              0
            </span>
          </h1>
          <button
            class="counter__add"
          >
            +
          </button>
          <button
            class="counter__decrement"
          >
            -
          </button>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.40",
  },
  "__setProps": [Function],
  "componentVM": {
    "count": 0,
    "decrement": [Function],
    "hasOwnProperty": [Function],
    "increment": [Function],
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <div
    class="counter"
  >
    <h1
      class="counter__text"
    >
       Current count is: 
      <span
        class="counter__count"
      >
        0
      </span>
    </h1>
    <button
      class="counter__add"
    >
      +
    </button>
    <button
      class="counter__decrement"
    >
      -
    </button>
  </div>,
}
`;
